lib/api_remote.ml

```diff
@@ -113,6 +100,72 @@ let get_permalink ~(ctx : Context.t) ~(req : Slack_t.get_permalink_req) =
     ~name:(sprintf "chat.getPermalink (%s, %s)" req.channel req.message_ts)
     ~ctx `GET api_path Slack_j.read_get_permalink_res
 
+let www_form_of_get_upload_url_ext (req : Slack_t.get_upload_url_ext_req) =
+  let fields =
+    [
+      Some ("filename", req.filename);
+      Some ("length", Int.to_string req.length);
+      string_field_val req.alt_txt "alt_txt";
+      string_field_val req.snippet_type "snippet_type";
+    ]
+  in
+  list_filter_opt fields
+
+let get_upload_url_external ~(ctx : Context.t) ~(req : Slack_t.get_upload_url_ext_req) =
+  log#info "getting upload url for %s" req.filename;
+  let args = www_form_of_get_upload_url_ext req in
+  let data = Web.make_url_args args in
+  let body = `Raw ("application/x-www-form-urlencoded", data) in
```

Comment:
if you want to use `application/x-www-form-urlencoded` here instead of `text/plain`, you should use \`Form instead of \`Raw